docs/building_pipelines.rst

```diff
@@ -293,3 +294,423 @@ Warnings raised will also be logged to the error log but will not cause data to
+                 "gearbox": 27.9788,
+                 "motor": 29.9785,
+                 "headstock": 23.65,
+                 "tailstock": 20.1666666667,
+                 "tool": 18.73
+               }
+             },
+             {
+               "asset_code": "lathe",
+               "user_ts": "2025-04-01 08:18:42.262955+00:00",
+               "ts": "2025-04-01 08:18:42.262955+00:00",
+               "reading": {
+                 "rpm": 349,
+                 "x": 0,
+                 "depth": 40,
+                 "state": "Spining Up"
+               }
+             }
+           ]
+         },
+     ...
+
+The example has been truncated at the second node merely to save space in the documentation.
+
+suspendIngest
+#############
+
+When the debugger is attached to a service it does not stop the service from ingesting and processing new data, it merely adds a way to view the latest data as it traverse the pipeline. In situation where the user wishes to troubleshot the operation of the pipeline it may be desirable to suspend the service from ingesting new data. This allows the data to be examined, configuration to be modified and data resent through the pipeline to observe the impact of configuration changes. To stop new data ingesting into the pipeline use the *suspendIngest* command.
+
+.. code-block:: bash
+
+   Debug: Lathe$ suspendIngest
+   {
+     "status": "ok"
+   }
+   Debug: Lathe$
+
+No new data will be read into the service, either by calling the poll entry point for a south service or fetching data from storage for a north service.
+
+.. note::
+
+   If the south plugin is an asynchronous plugin, any new data that the plugin tries to send into the pipeline will be discarded whilst the pipeline ingest is suspended.
+
+If trouble shooting a pipeline it is also useful to stop the pipeline send data out to the storage layer, in the case of a south plugin, or upstream to the destination system if a north plugin. This can be down using the *isolate* command.
+
+isolate
+#######
+
+Stops the pipeline emitting data into the storage layer, if a south service is attached to the debugger, or to the upstream system if the attached service is a north service.
+
+.. code-block:: bash
+
+   Debug: Lathe$ isolate
+   {
+     "status": "ok"
+   }
+   Debug: Lathe$ 
+
+state
+#####
+
+It is always possible to see the state of the debugger and the pipeline it is attached to by using the *state* command.
+
+.. code-block:: bash
+
+   Debug: Lathe$ state
+   {
+     "debugger": "Attached",
+     "ingress": "Suspended",
+     "egress": "Isolated"
+   }
+   Debug: Lathe$
+
+step
+####
+
+When a pipeline has its ingest suspended it can be useful to allow one or more new readings to be ingested to see the impact of any configuration changes on new data. This can be down using the *step* command. It can be passed an optional number of readings to ingest, if no number is passed then a single reading will be ingested.
```

Comment:
- This can be down using the *step* command -> This can be done using the *step* command
- Start a new sentence at "If no number is passed then a single reading..."